lib/Graph/Graph.cpp

```diff
@@ -621,16 +622,18 @@ std::string Module::getPrefix(llvm::StringRef name) {
 
 llvm::StringRef Module::uniqueName(llvm::StringRef name,
                                    const llvm::StringSet<> &stringTable,
-                                   llvm::StringSet<> &updateTable) {
+                                   llvm::StringSet<> &updateTable,
+                                   llvm::StringSet<> &originalNames) {
   std::string legalName = legalizeName(name);
   if (stringTable.find(legalName) == stringTable.end()) {
     auto it = updateTable.insert(legalName);
     if (it.second) {
       return it.first->first();
     }
   }
-
-  std::string prefix = Module::getPrefix(legalName);
+  std::string prefix = (originalNames.find(legalName) == originalNames.end())
```

Comment:
nit: perhaps add a comment here saying that we retain the `"__[0-9]+"` if it was in the original name?